cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/internals/BrokerAndSortedReplicas.java

```diff
@@ -36,6 +39,33 @@ public Broker broker() {
     return sortedReplicas;
   }
 
+  /**
+   * Get sorted replicas to move out of this broker, which are filtered based on the state of the cluster.
+   *
+   * If cluster has dead brokers, but this broker is alive, then replicas to move out are limited to the immigrants.
+   *
+   * @param clusterModel The state of the cluster.
+   * @return sorted replicas to move out of this broker, which are filtered based on the state of the cluster.
+   */
+  public List<Replica> replicasToMoveOut(ClusterModel clusterModel) {
+    List<Replica> replicasToMoveOut = new ArrayList<>(sortedReplicas());
```

Comment:
Instead of copying the whole sorted set and then prune, maybe it is more efficient that we scan the sorted set first and find the end element, then just copy a subset using `SortedSet<E> subSet(E fromElement, E toElement)`